src/backend/distributed/shared_library_init.c

```diff
@@ -234,6 +238,33 @@ _PG_init(void)
 }
 
 
+/*
+ * Stack size increase during high memory load may cause unexpected crashes.
+ * With this alloca call, we are increasing stack size explicitly, so that if
+ * it is not possible to increase stack size, we will get an OOM error instead
+ * of a crash.
+ *
+ * This function is called on backend startup. The allocated memory will
+ * automatically be released at the end of the function's scope. However, we'd
+ * have already expanded the stack and it wouldn't shrink back. So, in a sense,
+ * per backend we're securing max_stack_depth kB's of memory on the stack upfront.
+ *
+ * Not all the backends require max_stack_depth kB's on the stack, so we may
+ * unnecessary allocate. However, the default value is 2MB, which seems an
```

Comment:
allocate -> `we might end up with unnecessary allocations` ?

Reply:
```
@byucesoy I still think that we should do the performance tests that are in our release testing before back-porting it. We should at least do the pgbench tests, which are mostly short read/write queries, where this change might have some affect and we should see it before back-porting.
```

I don't see how this realistically could have a meaningful performance effect. The code should solely run once in postmaster (at startup), and after that it's going to make things minimally faster, not slower.